### PR TITLE
Refactor protocol manager 

### PR DESCRIPTION
**Pull request by tailingchen**

### CODE REVIEW

File: eth/pbft_handler.go [hunk 1]
```diff
@@ -617,9 +146,10 @@ func (pm *pbftProtocolManager) eventLoop() {
 
 // event loop for PBFT events
 func (pm *pbftProtocolManager) sendEvent(event pbft.ConsensusDataEvent) {
-	p := pm.peers.Peer(event.PeerID)
+	// FIXME: it's inefficient because retrieving all peers every time
```
Comment (yutelin) [inline eth/pbft_handler.go]: Create a ticket to track this.

File: eth/handler.go [hunk 2]
```diff
@@ -351,7 +351,10 @@ func (pm *protocolManager) handleMsg(p *peer) error {
 		return errResp(ErrMsgTooLarge, "%v > %v", msg.Size, ProtocolMaxMsgSize)
 	}
 	defer msg.Discard()
+	return handleMsg(msg, p)
+}
 
+func (pm *protocolManager) handleMsg(msg p2p.Msg, p *peer) error {
```
Comment (markya0616) [inline eth/handler.go]: suggest to use handleMsg(p *peer, msg p2p.Msg)

File: eth/pbft_handler.go (hunk 1 above)
Comment (tailingchen) [inline eth/pbft_handler.go]: ok

File: consensus/pbft/backends/simple/backend.go [hunk 3]
```diff
@@ -36,7 +36,6 @@ import (
 func New(config *pbft.Config, eventMux *event.TypeMux, privateKey *ecdsa.PrivateKey, db ethdb.Database) consensus.PBFT {
 	backend := &simpleBackend{
 		config:       config,
-		peerSet:      newPeerSet(),
```
Comment (bailantaotao) [inline consensus/pbft/backends/simple/backend.go]: valSet改成在這裡初始化而不是在Start()?

Comment (tailingchen) [inline consensus/pbft/backends/simple/backend.go]: valSet還是只能在Start()初始化, 因為會需要chain reader 才能算出最新的block的validator set

File: eth/handler.go (hunk 2 above)
Comment (tailingchen) [inline eth/handler.go]: fixed

File: consensus/pbft/backends/simple/backend.go (hunk 3 above)
Comment (bailantaotao) [inline consensus/pbft/backends/simple/backend.go]: OK

**Pull request merged**